atom/renderer/api/atom_api_web_frame.cc

```diff
@@ -226,6 +251,68 @@ void WebFrame::ExecuteJavaScript(const base::string16& code,
       callback.release());
 }
 
+void WebFrame::ExecuteJavaScriptInIsolatedWorld(int world_id,
+                                                const std::vector<mate::Dictionary> scripts,
+                                                mate::Arguments* args) {
+  std::vector<blink::WebScriptSource> sources;
+
+  for (int index = 0, size = scripts.size(); index < size; ++index) {
+    const mate::Dictionary script = scripts[index];
+    base::string16 code;
+    base::string16 url;
+    int start_line;
+    bool has_code = script.Get("code", &code);
+    bool has_url = script.Get("url", &url);
+    bool has_start_line = script.Get("startLine", &start_line);
+
+    if (!has_start_line)
+      start_line = 1;
+
+    if (has_code && has_url)
+      sources.push_back(blink::WebScriptSource(
```

Comment:
I think we can safely just do

```cpp
sources.emplace_back(blink::WebString::FromUTF16(code), GURL(url), start_line);
```

without any other checks `if` checks. `url` being empty should already be properly handled.